### ISSUE TITLE
Material Tracker Not Importing CSVs

### ISSUE THREAD

**Issue by twittyc** [Michael-R-R/GW2_Tool_Hub#8]
The Material tracker seems to fail to import the CSVs from gw2efficency.
I'm using the latest .zip install from your Github pages site.

Steps:

1. Export CSV from gw2efficency.
2. Go to Material Tracker tab in tool.
3. Click "Add Tab" and name the tab.
3. Click "Import Excel Sheet"
4. Select CSV file
5. Nothing happens

The status indicated on the bottom left shows the import to be successful. However, nothing appears in the Material Tracker tab.

Also I have tried a few times to import my API key (I verified the key is unrestricted.) When I click update Materials I get the error "Host requires authentication". I have tried the previous solution of reinstalling the application, but it does not work.

**Comment by Michael-R-R** [comment]
Hey there,

Thanks for contacting me about this issue. So, the first thing that I thought that could possibly be the issue is that you may not have excel installed on your PC, but I have went ahead and tested the application on a pc that doesn't have Excel and it worked, so I don't think that is the issue. 

After some testing one way I have been able to replicate both problems of the importing not working and the "Host Requires authentication" is by deleting the database that is suppose to be installed during installation. What I have noticed is that if the database is not installed for whatever reason then when you make a tab and do anything that requires querying the database it will actually make a database with the same name, "user.db" but it is a completely empty database and doesn't have the information that the Material Tracker tool needs.

If you don't mind, could you uninstall and do a fresh install and verify whether or not the database is being installed to your selected location, it should look like the following screenshot:

![troubleshoot](https://user-images.githubusercontent.com/54217603/121980799-738b2800-cd41-11eb-8ed8-5d4d3b24031d.png)

If that is the case and the database is not installing then you can grab the database from here: https://github.com/Michael-R-R/GW2_Tool_Hub/tree/main/NeededExtras and then just add the database to the installation folder and see if you still get the same errors.

**Comment by twittyc** [comment]
`user.db` does exist in my installation folder. I did a fresh install earlier and same issue.
I'm under the impression there may be some missing dependencies not being installed with the installer, but I'm not sure which.

![image](https://user-images.githubusercontent.com/17092044/121993847-70a02f80-cd61-11eb-98bd-c690241e1fd3.png)

**Comment by Michael-R-R** [comment]
I am not entirely sure what may be happening. I have tested it on my 3 PCs here at home and can't seem to replicate the errors unless deleting the database. I took a look at the inno script for compiling the .exe and everything looks good on that end. 

I will leave this issue open in case I can think of what else could be causing this issue.

**Comment by Michael-R-R** [comment]
I am going to investigate whether or not it is a issue with a missing dependency that happens to be on all my PCs, but may not be on others. Which like you mentioned in most likely the case here.

**Comment by Michael-R-R** [comment]
So did a little bit more testing this morning. Here are some things that I did:

- Downloaded and used Dependency in order to check if there were any missing dependencies, but it looked like everything checked out fine.
- Re-deployed and re-packaged the application in case there were any mistakes that I made during the process.
- Installed and created a virtual machine for Windows 10 64-bit in order to test the application on a fresh Windows 10 OS install
   - Somethings to note:
   - Everything installed fine, but since it was a fresh OS install I did have to install the vc_redist.x86.exe and vc_redist.x64.exe redistributables in order to run the application
   - The importing of excel sheets worked after adding a new tab (tab name was left as default), then importing a exported .csv excel sheet from GW2 Efficiency
   - Updating materials worked as intended too after adding the API key

Upon testing on the virtual machine with a fresh install of Windows 10 OS I had no errors with importing .csv excel sheets from GW2 Efficiency nor updating materials. Please feel free to try the new .zip installer download and let me know if you are still having the same issues.

Also keep in mind that installing to the Program Files/Program Files(x86) might be causing issues too as those locations can have restrictive access to read/write operations, causing the API key to not be written to the database. Try installing in a different location in order to make sure that you API key is being written to the database.

**Comment by Michael-R-R** [comment]
Another thing I just realized that is another issue is which excel sheet you are exporting from GW2 Efficiency, from as far as I can tell there are two different sheets (shopping-items and owned-materials). From my understanding the shopping-items shows you all the materials that you still need for the given recipe and owned-materials just shows you the materials that you have. Refer to the following screenshot and notice the differences.
![export](https://user-images.githubusercontent.com/54217603/122108872-96651d00-cdd1-11eb-9b6c-1328e6e4df9b.png)

Importing the wrong excel sheet is unintended behavior and will be corrected.

**Comment by twittyc** [comment]
Hmmm good question let me check.

**Comment by twittyc** [comment]
Aight so I did some testing. I moved the default install location out of program files and the API key is now working correctly. I also double checked the file I was importing and it was indeed the owned materials file. After using the shopping list file it worked successfully. I appreciate your time on this. This app looks really handy.

One thing to note is the default install location is in program files. That location will not work by default for some users. Might want to update permissions on the installed files or change the default install location.
